src/tools/compiletest/src/runtest.rs

```diff
@@ -263,7 +263,7 @@ impl<'test> TestCx<'test> {
 
     fn should_run_successfully(&self) -> bool {
         match self.config.mode {
-            RunPass => true,
+            RunPass => !self.props.skip_codegen,
```

Comment:
I'm not crazy about having `RunPass` be a distinct member of this enum, rather than just having it set flags on `self.props` (and/or adding appropriate flags to all the tests). e.g.,, does this mean that `skip_codegen` .. doesn't work on `// run-pass` UI tests, but does work on `run-pass` mode tests?

Reply:
I've been assuming that `self.props` was meant to reflect the actual set of directives on the header.

I can certainly change this code so that `// skip-codegen` works in UI tests too.

But yes, I would personally prefer if things were rearchitected so that the set of properties were established up front, based on a combination of the header (under current revision/compare-mode), and then queries could look solely at those properties.